config/application.rb

```diff
@@ -23,7 +23,13 @@ class Application < Rails::Application
 
     require "whitehall"
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 6.1
+    config.load_defaults 7.0
+
+    # Disable redirect protection. This is turned on by default in Rails 7.0.
+    config.action_controller.raise_on_open_redirects = false
+
+    # Disable rails 7.0+ button_to behaviour
+    config.action_view.button_to_generates_button_tag = false
```

Comment:
It'd be good to explain why these are switched off? Do we plan to have them switched off forever or is this just temporary?

Reply:
To be honest the latter would be simple enough to turn on, it would just mean updating tests to match the new generated output.

I'm not sure I understand enough about the former to know why it was necessary but it seems like it would be worth digging into, either as part of this card or a follow-up one.

Reply:
`raise_on_open_redirects` turns out to be a false positive. We're redirecting to an absolute URL generated with `Plek.new.website_root` which Rails detects as a third-party redirect (because it's absolute).

Reply:
Does we use those redirects everywhere or just in a few places? The preferential option would be to use the `allow_other_host` argument on the redirects where we expect this, but not practical if it's all of them.

Reply:
Yep, I've just pushed changes to address that and the other overridden default.

Reply:
OK, `button_to_generates_button_tag` seems to be more complex than I thought; it seems like there's somewhere in the code still generating `<input>` rather than `<button>`, and initially I'd thought maybe some could be left as-is but now I'm at a point where changing the test helpers breaks some tests, but leaving them breaks others. It may be possible to unpick it further though if I can figure out where the old-style buttons are coming from.